pkg/controller/configmap/config.go

```diff
@@ -316,6 +291,76 @@ func validateConfig(cfg *ini.File, sec string, key string) error {
 	return nil
 }
 
+// Validate node agent specific options
+func validateNodeAgentOptions(configmap *corev1.ConfigMap) []error {
+	errs := []error{}
+	data := &configmap.Data
+	cfg, err := ini.Load([]byte((*data)[operatortypes.ConfigMapDataKey]))
+
+	// Check MTU value
+	mtu := cfg.Section("nsx_node_agent").Key("mtu").Value()
+	_, err = strconv.Atoi(mtu)
+	if err != nil {
+		appendErrorIfNotNil(&errs, errors.Wrapf(err, "mtu is invalid"))
+	}
+	return errs
+}
+
+// Validate common options for both ncp and node agent
+func validateCommonOptions(configmap *corev1.ConfigMap) []error {
+	errs := []error{}
+	data := &configmap.Data
+	cfg, _ := ini.Load([]byte((*data)[operatortypes.ConfigMapDataKey]))
+	// Check DEFAULT section log_file option
+	if cfg.Section("DEFAULT").Key("log_file").Value() != "" {
+		appendErrorIfNotNil(&errs, errors.Errorf("DEFAULT section option: log_file is not allowed to change, current value:%s",
+			cfg.Section("DEFAULT").Key("log_file").Value()))
+	}
+	return errs
+}
+
+// Validate non supported options
+// Including WCP and TAS specific options, MP supported options
+func validateUnSupportedOptions(configmap *corev1.ConfigMap) []error {
+	errs := []error{}
+	data := &configmap.Data
+	is_changed := false
+	cfg, err := ini.Load([]byte((*data)[operatortypes.ConfigMapDataKey]))
+
+	for section, keys := range operatortypes.MPOptions {
+		for _, key := range keys {
+			if optionInConfigMap(configmap, section, key) {
+				log.Info(fmt.Sprintf("%s section option: %s is not supported in PolicyAPI, ignored this config option", section, key))
+				cfg.Section(section).DeleteKey(key)
+				is_changed = true
+			}
+		}
+	}
+
+	for section, keys := range operatortypes.WCPOptions {
+		for _, key := range keys {
+			if optionInConfigMap(configmap, section, key) {
+				log.Info(fmt.Sprintf("%s section option: %s is WCP specific option and not supported, ignored this config option", section, key))
```

Comment:
Again I feel it's not a good idea to expose any other platform in log or code, could we remove WCP here?
Maybe we can just merge MPOptions and WCPOptions to UnsupportedOptions?

Reply:
agreed. I remove these words from log but keep in code.